packages/aws-cdk-lib/aws-ecs/lib/base/base-service.ts

```diff
@@ -643,6 +651,14 @@ export abstract class BaseService extends Resource
       this.node.addDependency(taskDefinition);
     }
 
+    if (props.taskDefinitionRevision) {
+      this.resource.taskDefinition = taskDefinition.family;
+      if (props.taskDefinitionRevision !== 'latest') {
+        this.resource.taskDefinition += `:${props.taskDefinitionRevision}`;
+      }
+      this.node.addDependency(taskDefinition);
+    }
+
```

Comment:
```suggestion
    } else if (props.taskDefinitionRevision) {
      if (props.taskDefinitionRevision !== 'latest' && props.taskDefinitionRevision < 1) {
        throw new Error(`taskDefinitionRevision must be 'latest' or a positive number, got ${props.taskDefinitionRevision}`);
      }
      this.resource.taskDefinition = taskDefinition.family + (props.taskDefinitionRevision !== 'latest' 
 `:${props.taskDefinitionRevision}` : '');
      this.node.addDependency(taskDefinition);
    }

```
Should the revision number be enforced only for non-`CODE_DEPLOY` types?
Also, I would add validation for the revision when numeric (and a relative unit test).

Reply:
Hmmm... initially I didn't see any reason why the revision can't be made more specific than `latest` for `CODE_DEPLOY` - it's only as it is so that it's not _whatever the latest version is at the time of deployment_. However, the entire point of `CODE_DEPLOY` is that it's being deployed from code deploy, so if a new deployment is triggered it would override this anyways. I could maybe see a use case for if you want to temporarily pin and just not run the pipeline, but that seems rare and likely to cause more confusion than benefit. So you're probably right.

Good point on the additional validation

Reply:
> the entire point of CODE_DEPLOY is that it's being deployed from code deploy, so if a new deployment is triggered it would override this anyways

If this is the case it's better to keep the `else if` then, otherwise `this.resource.taskDefinition` would get overridden if `taskDefinitionRevision` is numeric.

Reply:
In this case, I'll also throw an error if a non-latest revision is specified and the deployment type is CODE_DEPLOY, to prevent confusion around explicitly asking for something different and it silently not doing that